tests/periph_i2c/main.c

```diff
@@ -20,270 +20,547 @@
 
 #include <stdio.h>
 #include <stdlib.h>
+#include <errno.h>
 
 #include "periph_conf.h"
 #include "periph/i2c.h"
 #include "shell.h"
 
+#ifndef GET_I2C_CLK_SPD
+#define GET_I2C_CLK_SPD(x)  (-1)
+#endif
+
+#ifndef I2C_ACK
+#define I2C_ACK  (0)
+#endif
+
+#define INVALID_ARGS    puts("Error: Invalid number of arguments");
+
 #define BUFSIZE        (128U)
 
 static int i2c_dev = -1;
 
-static i2c_speed_t map_speed[] = {
-    I2C_SPEED_LOW,
-    I2C_SPEED_NORMAL,
-    I2C_SPEED_FAST,
-    I2C_SPEED_FAST_PLUS,
-    I2C_SPEED_HIGH
-};
 
-int cmd_init_master(int argc, char **argv)
-{
-    int dev, speed, res;
+int check_dev(void) {
+    if (i2c_dev < 0) {
+        puts("Error: no I2C device was initialized");
+        return 1;
+    }
+    return 0;
+}
+
+int print_i2c_error(int res) {
+    if (res == -EOPNOTSUPP) {
+        printf("Error: EOPNOTSUPP [%d]\n", -res);
+        return 1;
+    }
+    else if (res == -EINVAL) {
+        printf("Error: EINVAL [%d]\n", -res);
+        return 1;
+    }
+    else if (res == -EAGAIN) {
+        printf("Error: EAGAIN [%d]\n", -res);
+        return 1;
+    }
+    else if (res == -ENXIO) {
+        printf("Error: ENXIO [%d]\n", -res);
+        return 1;
+    }
+    else if (res == -EIO) {
+        printf("Error: EIO [%d]\n", -res);
+        return 1;
+    }
+    else if (res == -ETIMEDOUT) {
+        printf("Error: ETIMEDOUT [%d]\n", -res);
+        return 1;
+    }
+    else if (res == I2C_ACK) {
+        printf("Success: I2C_ACK [%d]\n", res);
+        return 0;
+    }
+    puts("Error: Unknown Error");
+    return 1;
+}
+
+int cmd_i2c_assign_dev(int argc, char **argv) {
+    int dev;
 
-    if (argc != 3) {
-        puts("Error: Init: Invalid number of arguments!");
-        printf("Usage:\n%s: [DEVICE] [SPEED]\n", argv[0]);
+    if (argc != 2) {
+        INVALID_ARGS;
+        printf("Usage:\n%s: [DEVICE]\n", argv[0]);
         puts("    with DEVICE:");
         for (unsigned i = 0; i < I2C_NUMOF; i++) {
-            printf("          %u -> I2C_%u\n", i, i);
+            printf("          %u -> i2c_%u\n", i, i);
         }
-        puts("         SPEED:");
-        puts("          0 -> SPEED_LOW (10kbit/s)");
-        puts("          1 -> SPEED_NORMAL (100kbit/s)");
-        puts("          2 -> SPEED_FAST (400kbit/s)");
-        puts("          3 -> SPEED_FAST_PLUS (1Mbit/s)");
-        puts("          4 -> SPEED_HIGH (3.4Mbit/s)\n");
         return 1;
     }
-
     dev = atoi(argv[1]);
-    speed = atoi(argv[2]);
+    if (dev < 0 || dev >= (int)I2C_NUMOF) {
```

Comment:
minor nit, but this can be simplified to:
```
if (dev < 0 || dev >= (int)I2C_NUMOF) {
    puts("Error: Given device not available");
    i2c_dev = -1;
    return 1;
}
printf("Success: i2c_%i successfully assigned as master\n", dev);
i2c_dev = dev;
return 0;
```

omitting the `else` branch because of the `return` in the `if` part and also the final `return 1` would've never been reached, too.

Reply:
Yup, will do. [TODO]